test/e2e/windows/volumes.go

```diff
@@ -0,0 +1,172 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"fmt"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	emptyDirVolumePath = "C:\\test-volume"
+	hostMapPath        = "C:\\tmp"
+	containerName      = "test-container"
+	volumeName         = "test-volume"
+)
+
+var (
+	image = imageutils.GetE2EImage(imageutils.Pause)
+)
+
+var _ = SIGDescribe("Windows volume mounts ", func() {
+	f := framework.NewDefaultFramework("windows-volumes")
+	var (
+		emptiDirSource = v1.VolumeSource{
+			EmptyDir: &v1.EmptyDirVolumeSource{
+				Medium: v1.StorageMediumDefault,
+			},
+		}
+
+		hostMapSource = v1.VolumeSource{
+			HostPath: &v1.HostPathVolumeSource{
+				Path: hostMapPath,
+			},
+		}
+	)
+	BeforeEach(func() {
+		framework.SkipUnlessNodeOSDistroIs("windows")
+	})
+
+	Context("check volume mount permissions", func() {
+
+		It("container should have readOnly permissions on emptyDir", func() {
+
+			By("creating a container with readOnly permissions on emptyDir volume")
+			doReadOnlyTest(f, emptiDirSource, emptyDirVolumePath)
+
+			By("creating two containers, one with readOnly permissions the other with read-write permissions on emptyDir volume")
+			doReadWriteReadOnlyTest(f, emptiDirSource, emptyDirVolumePath)
+		})
+
+		It("container should have readOnly permissions on hostMapPath", func() {
+
+			By("creating a container with readOnly permissions on hostMap volume")
+			doReadOnlyTest(f, hostMapSource, hostMapPath)
+
+			By("creating two containers, one with readOnly permissions the other with read-write permissions on hostMap volume")
+			doReadWriteReadOnlyTest(f, hostMapSource, hostMapPath)
+		})
+
+	})
+
+})
+
+func doReadOnlyTest(f *framework.Framework, source v1.VolumeSource, volumePath string) {
+	var (
+		filePath = volumePath + "\\test-file.txt"
+		podName  = "pod-" + string(uuid.NewUUID())
+		pod      = testPodWithROVolume(podName, source, volumePath)
+	)
+
+	f.PodClient().CreateSync(pod)
+	cmd := []string{"cmd", "/c", "echo windows-volume-test", ">", filePath}
+
+	_, stderr, _ := f.ExecCommandInContainerWithFullOutput(podName, containerName, cmd...)
+
+	Expect(stderr).To(Equal("Access is denied."))
```

Comment:
Can you also check that read access works?

Reply:
we've had bugs where you couldn't read or write before

Reply:
This goes back to the question above, read access is being checked in the two container test since the other container writes to a file that the other can read. Do we write a different test to check read access of a different non empty volume?

Reply:
@PatrickLang thoughts on this?